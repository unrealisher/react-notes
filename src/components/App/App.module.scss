.body {
  width: 100%;
  min-height: 100vh;
  display: grid;
  grid-template-rows: 64px auto 64px;
}

.header {
  grid-row: 1/2;
}

.main {
  box-sizing: border-box;
  grid-row: 2/3;
  padding: 16px;
  padding-bottom: 0;
  background-color: #f7f7f7;

  @media (min-width: 1366px) {
    padding: 32px 173px;
  }
}

.footer {
  grid-row: 3/4;
}
