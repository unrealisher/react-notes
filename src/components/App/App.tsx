import React, { useEffect, useState } from "react";
import { connect } from "react-redux";

import Header from "./../Header/Header";
import Footer from "./../Footer/Footer";
import Colors from "./../Colors/Colors";
import Notes from "./../Notes/Notes";

import styles from "./App.module.scss";
import { fetchData } from "../../store/actions/fetchData";
import INote from "../../interfaces/INote";
import NoteConstructor from "../NoteConstructor/NoteConstructor";
import { ThunkDispatch } from "redux-thunk";
import IState from "../../interfaces/IState";
import { AnyAction } from "redux";

interface IDispatchToProps {
  onFetchData: Function;
}

interface IProps extends IDispatchToProps {}

export const App = (props: IProps): JSX.Element => {
  const { onFetchData } = props;
  const [popup, setPopup] = useState<boolean>(false);
  const [patchItem, setPatchItem] = useState<INote>({});

  useEffect(() => {
    onFetchData();
  });

  return (
    <div className={styles.body}>
      <header className={styles.header}>
        <Header setPopup={setPopup} />
      </header>
      <main className={styles.main}>
        <Colors />
        <Notes setPatchItem={setPatchItem} setPopup={setPopup} />
      </main>
      <footer className={styles.footer}>
        <Footer />
      </footer>
      {popup && (
        <NoteConstructor
          setPopup={setPopup}
          setPatchItem={setPatchItem}
          note={patchItem}
        />
      )}
    </div>
  );
};

const mapDispatchToProps = (
  dispatch: ThunkDispatch<IState, null, AnyAction>
): IDispatchToProps => {
  return {
    onFetchData: (): void => {
      dispatch(fetchData());
    }
  };
};

export default connect(
  null,
  mapDispatchToProps
)(App);
