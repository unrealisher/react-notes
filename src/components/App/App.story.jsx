import React from "react";
import { Provider } from "react-redux";
import store from "./../../store/index";

import App from "./App";

import { storiesOf } from "@storybook/react";

import "./../../index.scss";

storiesOf("App", module).add("Стандарт", () => (
  <Provider store={store}>
    <App />
  </Provider>
));
