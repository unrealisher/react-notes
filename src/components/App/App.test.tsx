import React from "react";
import { shallow } from "enzyme";
import { App } from "./App";

describe("App", () => {
  it("correct render", () => {
    const component = shallow(<App onFetchData={() => {}} />);
    expect(component).toMatchSnapshot();
  });
});
