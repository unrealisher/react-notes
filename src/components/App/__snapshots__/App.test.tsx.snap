// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App correct render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    onFetchData={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header
          className="header"
        >
          <Header
            setPopup={[Function]}
          />
        </header>,
        <main
          className="main"
        >
          <Connect(Colors) />
          <Connect(Notes)
            setPatchItem={[Function]}
            setPopup={[Function]}
          />
        </main>,
        <footer
          className="footer"
        >
          <Footer />
        </footer>,
        false,
      ],
      "className": "body",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Header
            setPopup={[Function]}
          />,
          "className": "header",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "setPopup": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Connect(Colors) />,
            <Connect(Notes)
              setPatchItem={[Function]}
              setPopup={[Function]}
            />,
          ],
          "className": "main",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "setPatchItem": [Function],
              "setPopup": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "main",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Footer />,
          "className": "footer",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "footer",
      },
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header
            className="header"
          >
            <Header
              setPopup={[Function]}
            />
          </header>,
          <main
            className="main"
          >
            <Connect(Colors) />
            <Connect(Notes)
              setPatchItem={[Function]}
              setPopup={[Function]}
            />
          </main>,
          <footer
            className="footer"
          >
            <Footer />
          </footer>,
          false,
        ],
        "className": "body",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Header
              setPopup={[Function]}
            />,
            "className": "header",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "setPopup": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Connect(Colors) />,
              <Connect(Notes)
                setPatchItem={[Function]}
                setPopup={[Function]}
              />,
            ],
            "className": "main",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "setPatchItem": [Function],
                "setPopup": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "main",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Footer />,
            "className": "footer",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "footer",
        },
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
